vermouth/processors/sort_molecule_atoms.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2018 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+from .processor import Processor
+
+
+class SortMoleculeAtoms(Processor):
+    """
+    Sort the atoms within a molecule by chain, resid, and resname.
+
+    This is usefull, for instance, when atoms have been added (*e.g.* missing
+    atoms identified by :class:`vermouth.processors.repair_graph.RepairGraph`).
+    The atom keys are left identical, only the order of the nodes is changed.
+    """
+    def run_molecule(self, molecule):
+        node_order = sorted(molecule, key=partial(_keyfunc, molecule))
+        sorted_nodes = [
+            (node_key, molecule.nodes[node_key])
+            for node_key in node_order
+        ]
+        # remove_nodes_from would be better, but it leads to an OrderedDict
+        # being modified during iteration, and python does not like that.
+        for node in node_order:
+            super(type(molecule), molecule).remove_node(node)
+        molecule.add_nodes_from(sorted_nodes)
```

Comment:
`nx.relabel_nodes(molecule, dict(sorted_nodes))` (or the inverse of sorted_nodes. You'll need to check it makes a Molecule again though. Also, what about interactions and such? In addition, what you're doing here loses all edges.

Reply:
From what I understand, `relabel_nodes` changes the key/labels but not the order. I want to do the opposite: changing the order but not the labels. I did not notice the issue with the edges, though. If the labels are not changed, then I can back the edges up, are recreate them.

Reply:
My bad.
`molecule.nodes.move_to_end`? Looks to be a lot easier.
https://docs.python.org/3/library/collections.html#collections.OrderedDict.move_to_end

Also, I glossed over line 36, but that's also something I'm not happy with at all.

Reply:
I'm still unhappy with line 41 (was 36).
It can not be a good idea to have to use `super` outside of an inheriting class.

Reply:
`molecule.nodes.move_to_end`, if possible.
Otherwise, just call remove_node (or better, remove_nodes_from, can't see what that doesn't work) and re-add the interactions.